### DISCUSSION TITLE
Nano Pi M1 Plus

### DISCUSSION THREAD

**Discussion by k6bpm** [adafruit/Adafruit_Blinka#1080] [New Board Requests]
Hi All,

I need to update a project I created a few years ago for a data acquisition project for  work. The whole project needs to be refreshed and I need to build 5 of these right away.  Probably many more. I cannot find RPis *anywhere* these days unless I'm willing to pay lots of $$.  I have a couple of NanoPi M1 Plus boards I ordered a couple of years ago to evaluate and more stock is available. They are GPIO compatible with RPi's. I have everything set up and updated and the last step is to attach an I2C SSD1306 128x64 OLED display. Whoops! I updated all my Python libraries to support Blinka but it looks like my board is not supported. I like these boards and would like to use them, but I have to have the SSD1306 working. 

Thanks, any help would very much appreciated. Details are shown below:

** OS 

Ubuntu (core) 20.04.4 LTS 4.14.111 (FrendlyELEC)

** cat /proc/cpuinfo output

processor	: 0
model name	: ARMv7 Processor rev 5 (v7l)
BogoMIPS	: 22.85
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xc07
CPU revision	: 5

processor	: 1
model name	: ARMv7 Processor rev 5 (v7l)
BogoMIPS	: 22.85
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xc07
CPU revision	: 5

processor	: 2
model name	: ARMv7 Processor rev 5 (v7l)
BogoMIPS	: 22.85
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xc07
CPU revision	: 5

processor	: 3
model name	: ARMv7 Processor rev 5 (v7l)
BogoMIPS	: 22.85
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xc07
CPU revision	: 5

Hardware	: sun8i
Revision	: 0000
Serial		: 02c000810adc3f72

** blinkatest.py output

NotImplementedError: Adafruit-PlatformDetect version 3.24.1 was unable to identify the board and/or microcontroller running the Linux platform. Please be sure you have the latest packages running: 'pip3 install --upgrade adafruit-blinka adafruit-platformdetect'

** boards.py listings for Nano Pi (nothing for M1 or M1 Plus)

# Nano Pi boards
NANOPI_NEO_AIR = "NANOPI_NEO_AIR"
NANOPI_DUO2 = "NANOPI_DUO2"
NANOPI_NEO = "NANOPI_NEO"

** I2C and SPI Enabled:

crw-rw---- 1 root i2c   89, 1 Apr 21 12:54 /dev/i2c-1
crw-rw---- 1 root i2c   89, 3 Apr 21 12:54 /dev/i2c-3
crw------- 1 root root 153, 0 Apr 21 12:54 /dev/spidev0.0

** Python version:

Python 3.8.10

** pip3 list

pip3 list
Package                          Version
-------------------------------- --------------------
Adafruit-Blinka                  8.0.1
adafruit-circuitpython-busdevice 5.1.9
adafruit-circuitpython-framebuf  1.4.10
adafruit-circuitpython-ssd1306   2.12.5
adafruit-circuitpython-typing    1.7.1
Adafruit-GPIO                    1.0.3
Adafruit-PlatformDetect          3.24.1
Adafruit-PureIO                  1.1.9
Adafruit-SSD1306                 1.6.2
certifi                          2019.11.28
chardet                          3.0.4
configobj                        5.0.6
configparser                     5.2.0
dbus-python                      1.2.16
distro                           1.4.0
distro-info                      0.23ubuntu1
fail2ban                         0.11.1
idna                             2.8
ipaddress                        1.0.23
netifaces                        0.10.4
nmap                             0.0.1
pathlib                          1.0.1
Pillow                           9.1.1
pip                              22.1.2
pyftdi                           0.54.0
pygeoip                          0.3.2
PyGObject                        3.36.0
pyinotify                        0.9.6
pymacaroons                      0.13.0
PyNaCl                           1.3.0
pyserial                         3.5
python-apt                       2.0.0+ubuntu0.20.4.7
pyusb                            1.2.1
PyYAML                           5.3.1
requests                         2.22.0
setuptools                       45.2.0
six                              1.14.0
smbus2                           0.4.2
spidev                           3.5
ssh-import-id                    5.10
systemd-python                   234
ubuntu-advantage-tools           20.3
ufw                              0.36
urllib3                          1.25.8
wheel                            0.34.2

**Comment by ladyada** [comment]
you could probably follow our guides to add support - since it looks like I2C and SPI is already enabled
https://learn.adafruit.com/adding-a-single-board-computer-to-platformdetect-for-blinka
https://learn.adafruit.com/adding-a-single-board-computer-to-blinka
